### PR TITLE
[Bugfix] Initialization of group checkboxes based on Lizmap configuration

### LINKED ISSUE #3968: [Bug]: "Toggle" setting in Layers tab on Lizmap QGIS plugin not working on groups
### What is the bug?

Hi,
the follow configuration on Lizmap plugin:

group1

- group1 -> `Toggle` unchecked
- all children of group1 -> `Toggle` checked

seems not to be interpreted correctly by LWC

group1_LWC

The tree should be initialized like this (?)

group1_desired

Thanks

### Steps to reproduce the issue

See attached project to reproduce the issue:

[groups.zip](https://github.com/3liz/lizmap-web-client/files/13378213/groups.zip)

### Versions

**Versions :**

* Lizmap Web Client : 3.7.0-beta.1
* Lizmap plugin : 3.17.0
* QGIS Desktop : 3.28.7
* QGIS Server : 3.28.5
* Py-QGIS-Server : 
* QGIS Server plugin atlasprint : not found
* QGIS Server plugin lizmap_server : 2.8.1
* QGIS Server plugin wfsOutputExtension : 1.7.0

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### QGIS server version, only if the section above doesn't mention the QGIS Server version

_No response_

### Operating system

Debian

### Browsers

Firefox, Chrome

### Browsers version

last

### Relevant log output

_No response_

### LINKED ISSUE #4232: Layer toggeling behaviour at LWC start
### What is the bug? (in English)

If I understand it correctly, with LWC 3.7 the behaviour of the layer panel should be the same as in QGIS. This is really great and works fine - but "initiating" the layers which should be activated at the start of LWC does somehow not really follow an understandable logic (at least i can not comprehend it). Maybe it is intended to work this way, hopefully it is possible to clarify a little bit:

 

### Steps to reproduce the issue

1. When not working with QGIS map themes, setting a layer in the lizmap plugin as "toggled" will also toggle the group the layer is part of when opening LWC - although the group is _not_ toggled in the lizmap plugin (so all layers of that group need to be "untoggled" in LM plugin, but then the user has to activate all layers manually in LWC)
2. To circumvent this behaviour, I tried using QGIS map themes - toggle all layers in a group but do not toggle the group itself:
grafik
This results in:
grafik
which is again not the desired result.

So how can it be achieved that a user can "fast switch on" a group like shown here:

Group_SwitchOn

### Versions, safeguards, check summary etc

QGIS Desktop 3.28
LWC 3.7.2

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### Operating system

Ubuntu 22.04

### Browsers

Firefox, Chrome, Safari, Microsoft Edge

### Browsers version

Firefox 115

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mind84**
Groups, at any level, will be initialized according to `Lizmap` plugin configuration.

image

This topic has generated a few tickets/discussions, and since a lot of work has been done on groups lately, I've tried to pay attention to this to preserve any fixes made thus far (referring to PR https://github.com/3liz/lizmap-web-client/pull/4622 https://github.com/3liz/lizmap-web-client/pull/4618 )

The main reason why I want to propose this fix is ​​that the `Permalink` functionality already supports this type of initialization and I basically couldn't find any valid reason not to support this behavior on clean map initialization.

**TECHNICAL NOTES:**

- As said, all "group checkboxes" fix done so far should be preserved
- These changes affected many test files (`unit` tests and `e2e` tests), which were updated accordingly.
- I had to take into account the `hide checkboxes for groups` option to force the groups to be checked if checkbox is hided, despite the configuration. Since this option is not layer-specific, I had to pass the `option config object` through the `layertree` initialization methods and this also generated a lot of changes in the test files. However, passing `option` object might come in handy if there are new "global" options on the layertree in the future.
- Changes affects only map intialization. User interaction after map init is not changed (e.g. if user turn on al "sub layer", then corresponding parent group will be activated)

Ticket : #3968, #4232

fixes #3968, fixes #4232

Funded by Faunalia

### CODE REVIEW

File: assets/src/modules/State.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class State extends EventDispatcher {
      */
     get layersAndGroupsCollection() {
         if (this._collection == null) {
-            this._collection = new LayersAndGroupsCollection(this._initialConfig.layerTree, this._initialConfig.layersOrder);
+            this._collection = new LayersAndGroupsCollection(this._initialConfig.layerTree, this._initialConfig.layersOrder, this._initialConfig.options);
```
Comment (rldhont) [inline assets/src/modules/State.js:63]:
```suggestion
            this._collection = new LayersAndGroupsCollection(this._initialConfig.layerTree, this._initialConfig.layersOrder, this._initialConfig.options.hideGroupCheckbox);
```

File: assets/src/modules/state/Layer.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ import { LayerConfig } from './../config/Layer.js';
 import { AttributionConfig } from './../config/Attribution.js'
 import { LayerStyleConfig, LayerGeographicBoundingBoxConfig, LayerBoundingBoxConfig, LayerTreeGroupConfig, LayerTreeItemConfig } from './../config/LayerTree.js';
 import { buildLayerSymbology, LayerSymbolsSymbology, LayerIconSymbology, LayerGroupSymbology } from './Symbology.js';
+import { OptionsConfig } from '../config/Options.js';
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:17]:
```suggestion
```

File: assets/src/modules/state/Layer.js [hunk 3]
```diff
@@ -26,9 +27,10 @@ export class LayerItemState extends EventDispatcher {
      * Creating a layer item state
      * @param {string} type                          - the layer item type
      * @param {LayerTreeItemConfig} layerTreeItemCfg - the layer item config
+     * @param {OptionsConfig}       options          - the option instance from lizMap configuration
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:30]:
```suggestion
```

File: assets/src/modules/state/Layer.js [hunk 4]
```diff
@@ -26,9 +27,10 @@ export class LayerItemState extends EventDispatcher {
...
      * @param {LayerItemState}      [parentGroup]    - the parent layer group
      */
-    constructor(type, layerTreeItemCfg, parentGroup) {
+    constructor(type, layerTreeItemCfg, options, parentGroup) {
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:33]:
```suggestion
    constructor(type, layerTreeItemCfg, parentGroup) {
```

File: assets/src/modules/state/Layer.js [hunk 5]
```diff
@@ -553,10 +556,11 @@ export class LayerLayerState extends LayerItemState {
      * Creating a layer state
      * @param {LayerTreeItemConfig} layerTreeItemCfg - the layer item config
      * @param {number[]}            layersOrder      - the layers order
+     * @param {OptionsConfig}       options          - the option instance from lizMap configuration
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:559]:
```suggestion
```

File: assets/src/modules/state/Layer.js [hunk 6]
```diff
@@ -553,10 +556,11 @@ export class LayerLayerState extends LayerItemState {
...
      * @param {LayerGroupState}     [parentMapGroup] - the parent layer group
      */
-    constructor(layerTreeItemCfg, layersOrder, parentMapGroup) {
-        super('layer', layerTreeItemCfg, parentMapGroup);
+    constructor(layerTreeItemCfg, layersOrder, options, parentMapGroup) {
+        super('layer', layerTreeItemCfg, options, parentMapGroup);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:563]:
```suggestion
    constructor(layerTreeItemCfg, layersOrder, parentMapGroup) {
        super('layer', layerTreeItemCfg, parentMapGroup);
```

File: assets/src/modules/state/Layer.js [hunk 7]
```diff
@@ -739,10 +743,11 @@ export class LayerVectorState extends LayerLayerState {
      * Creating a vector layer state
      * @param {LayerTreeItemConfig} layerTreeItemCfg - the layer item config
      * @param {number[]}            layersOrder      - the layers order
+     * @param {OptionsConfig}       options          - the option instance from lizMap configuration
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:746]:
```suggestion
```

File: assets/src/modules/state/Layer.js [hunk 8]
```diff
@@ -739,10 +743,11 @@ export class LayerVectorState extends LayerLayerState {
...
      * @param {LayerGroupState}     [parentMapGroup] - the parent layer group
      */
-    constructor(layerTreeItemCfg, layersOrder, parentMapGroup) {
-        super(layerTreeItemCfg, layersOrder, parentMapGroup)
+    constructor(layerTreeItemCfg, layersOrder, options, parentMapGroup) {
+        super(layerTreeItemCfg, layersOrder, options, parentMapGroup)
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:750]:
```suggestion
    constructor(layerTreeItemCfg, layersOrder, parentMapGroup) {
        super(layerTreeItemCfg, layersOrder, parentMapGroup)
```

File: assets/src/modules/state/Layer.js [hunk 9]
```diff
@@ -1128,10 +1133,11 @@ export class LayerRasterState extends LayerLayerState {
      * Creating a raster layer state
      * @param {LayerTreeItemConfig} layerTreeItemCfg - the layer item config
      * @param {string[]}            layersOrder      - the layers order
+     * @param {OptionsConfig}       options          - the option instance from lizMap configuration
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1136]:
```suggestion
```

File: assets/src/modules/state/Layer.js [hunk 10]
```diff
@@ -1128,10 +1133,11 @@ export class LayerRasterState extends LayerLayerState {
...
      * @param {LayerGroupState}     [parentMapGroup] - the parent layer group
      */
-    constructor(layerTreeItemCfg, layersOrder, parentMapGroup) {
-        super(layerTreeItemCfg, layersOrder, parentMapGroup)
+    constructor(layerTreeItemCfg, layersOrder, options, parentMapGroup) {
+        super(layerTreeItemCfg, layersOrder, options, parentMapGroup)
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1140]:
```suggestion
    constructor(layerTreeItemCfg, layersOrder, parentMapGroup) {
        super(layerTreeItemCfg, layersOrder, parentMapGroup)
```

File: assets/src/modules/state/Layer.js [hunk 11]
```diff
@@ -44,7 +46,8 @@ export class LayerItemState extends EventDispatcher {
         this._geographicBoundingBox = null;
         this._minScaleDenominator = null;
         this._maxScaleDenominator = null;
-        this._checked = this._parentGroup == null ? true : false;
+        // prop checked is inherithed from Lizmap configuration. if checkboxes are hided for groups then they are checked by default
+        this._checked = this._parentGroup == null ? true : (type == 'group' && options.hideGroupCheckbox ? true : this._layerTreeItemCfg.layerConfig.toggled);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:50]:
```suggestion
        // prop checked is inherithed from Lizmap configuration
        this._checked = this._parentGroup == null ? true : this._layerTreeItemCfg.layerConfig.toggled;
```

File: assets/src/modules/state/Layer.js [hunk 12]
```diff
@@ -1165,10 +1171,11 @@ export class LayerGroupState extends LayerItemState {
      * Creating a layer group state
      * @param {LayerTreeGroupConfig} layerTreeGroupCfg - the layer item config
      * @param {number[]}             layersOrder       - the layers order
+     * @param {OptionsConfig}       options            - the option instance from lizMap configuration
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1174]:
```suggestion
     * @param {boolean}       hideGroupCheckbox            - the hideGroupCheckbox option instance from lizMap configuration
```

File: assets/src/modules/state/Layer.js [hunk 13]
```diff
@@ -1165,10 +1171,11 @@ export class LayerGroupState extends LayerItemState {
...
      * @param {LayerGroupState}     [parentMapGroup]  - the parent layer group
      */
-    constructor(layerTreeGroupCfg, layersOrder, parentMapGroup) {
-        super('group', layerTreeGroupCfg, parentMapGroup);
+    constructor(layerTreeGroupCfg, layersOrder, options, parentMapGroup) {
+        super('group', layerTreeGroupCfg, options, parentMapGroup);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1178]:
```suggestion
    constructor(layerTreeGroupCfg, layersOrder, hideGroupCheckbox, parentMapGroup) {
        super('group', layerTreeGroupCfg, parentMapGroup);
        // if checkboxes are hided for groups then they are checked by default
        if (hideGroupCheckbox) {
            this._checked = true;
        }
```

File: assets/src/modules/state/Layer.js [hunk 14]
```diff
@@ -1180,7 +1187,7 @@ export class LayerGroupState extends LayerItemState {
             // Group as group
             if (layerTreeItem instanceof LayerTreeGroupConfig) {
                 // Build group
-                const group = new LayerGroupState(layerTreeItem, layersOrder, this);
+                const group = new LayerGroupState(layerTreeItem, layersOrder, options, this);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1190]:
```suggestion
                const group = new LayerGroupState(layerTreeItem, layersOrder, hideGroupCheckbox, this);
```

File: assets/src/modules/state/Layer.js [hunk 15]
```diff
@@ -1195,17 +1202,13 @@ export class LayerGroupState extends LayerItemState {
...
-                // Group is checked if one child is checked
-                if (group.checked) {
-                    this._checked = true;
-                }
             } else if (cfg.type != 'group') {
                 // layer with geometry is vector layer
                 let layer = null;
                 if (cfg.geometryType != null) {
-                    layer = new LayerVectorState(layerTreeItem, layersOrder, this);
+                    layer = new LayerVectorState(layerTreeItem, layersOrder, options, this);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1209]:
```suggestion
                    layer = new LayerVectorState(layerTreeItem, layersOrder, this);
```

File: assets/src/modules/state/Layer.js [hunk 16]
```diff
@@ -1195,17 +1202,13 @@ export class LayerGroupState extends LayerItemState {
...
                 } else {
-                    layer = new LayerRasterState(layerTreeItem, layersOrder, this);
+                    layer = new LayerRasterState(layerTreeItem, layersOrder, options, this);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1211]:
```suggestion
                    layer = new LayerRasterState(layerTreeItem, layersOrder, this);
```

File: assets/src/modules/state/Layer.js [hunk 17]
```diff
@@ -1399,10 +1398,11 @@ export class LayersAndGroupsCollection extends EventDispatcher {
      * Creating the collection of layers and groups state
      * @param {LayerTreeGroupConfig} layerTreeGroupCfg - the layer item config
      * @param {number[]}             layersOrder       - the layers order
+     * @param {OptionsConfig}        options           - the option instance from lizMap configuration
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1401]:
```suggestion
     * @param {boolean}        hideGroupCheckbox           - the hideGroupCheckbox option instance from lizMap configuration
```

File: assets/src/modules/state/Layer.js [hunk 18]
```diff
@@ -1399,10 +1398,11 @@ export class LayersAndGroupsCollection extends EventDispatcher {
...
      */
-    constructor(layerTreeGroupCfg, layersOrder) {
+    constructor(layerTreeGroupCfg, layersOrder, options) {
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1403]:
```suggestion
    constructor(layerTreeGroupCfg, layersOrder, hideGroupCheckbox) {
```

File: assets/src/modules/state/Layer.js [hunk 19]
```diff
@@ -1399,10 +1398,11 @@ export class LayersAndGroupsCollection extends EventDispatcher {
...
         super();
-        this._root = new LayerGroupState(layerTreeGroupCfg, layersOrder);
+        this._root = new LayerGroupState(layerTreeGroupCfg, layersOrder, options);
```
Comment (rldhont) [inline assets/src/modules/state/Layer.js:1405]:
```suggestion
        this._root = new LayerGroupState(layerTreeGroupCfg, layersOrder, hideGroupCheckbox);
```

File: tests/js-units/node/state/baselayer.test.js [hunk 20]
```diff
@@ -43,7 +44,8 @@ function getBaseLayersState(name) {
 
     const layersOrder = buildLayersOrder(config, rootCfg);
 
-    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/baselayer.test.js:48]:
```suggestion
    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/baselayer.test.js [hunk 21]
```diff
@@ -168,7 +170,7 @@ describe('BaseLayersState', function () {
 
         const layersOrder = buildLayersOrder(config, rootCfg);
 
-        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/baselayer.test.js:173]:
```suggestion
        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/externallayertree.test.js [hunk 22]
```diff
@@ -39,7 +40,8 @@ function getRootLayerTreeGroupState(name) {
 
     const layersOrder = buildLayersOrder(config, rootCfg);
 
-    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/externallayertree.test.js:44]:
```suggestion
    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/externalmaplayer.test.js [hunk 23]
```diff
@@ -38,7 +39,8 @@ function getRootMapGroupState(name) {
 
 	const layersOrder = buildLayersOrder(config, rootCfg);
 
-	const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/externalmaplayer.test.js:43]:
```suggestion
        const options = new OptionsConfig(config.options);
        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/layer.test.js [hunk 24]
```diff
@@ -36,7 +37,8 @@ function getRootLayerGroupState(name) {
 
     const layersOrder = buildLayersOrder(config, rootCfg);
 
-    const root = new LayerGroupState(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    const root = new LayerGroupState(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/layer.test.js:41]:
```suggestion
    const root = new LayerGroupState(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/layer.test.js [hunk 25]
```diff
@@ -66,7 +68,8 @@ function getLayersAndGroupsCollection(name) {
 
     const layersOrder = buildLayersOrder(config, rootCfg);
 
-    return new LayersAndGroupsCollection(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    return new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/layer.test.js:72]:
```suggestion
    return new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/layer.test.js [hunk 26]
```diff
@@ -1616,7 +1623,8 @@ describe('LayersAndGroupsCollection', function () {
 
         const layersOrder = buildLayersOrder(config, rootCfg);
 
-        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+        const options = new OptionsConfig(config.options);
+        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/layer.test.js:1627]:
```suggestion
        const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/layerTree.test.js [hunk 27]
```diff
@@ -40,7 +41,8 @@ function getRootLayerTreeGroupState(name) {
 
     const layersOrder = buildLayersOrder(config, rootCfg);
 
-    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/layerTree.test.js:45]:
```suggestion
    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

File: tests/js-units/node/state/maplayer.test.js [hunk 28]
```diff
@@ -36,7 +37,8 @@ function getRootMapGroupState(name) {
 
     const layersOrder = buildLayersOrder(config, rootCfg);
 
-    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder);
+    const options = new OptionsConfig(config.options);
+    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options);
```
Comment (rldhont) [inline tests/js-units/node/state/maplayer.test.js:41]:
```suggestion
    const collection = new LayersAndGroupsCollection(rootCfg, layersOrder, options.hideGroupCheckbox);
```

**Review by rldhont — CHANGES_REQUESTED** [review]
It is not necessary to pass the entire `Options` object for one variable, and because it is `group` specific the default checked value based on `hideGroupCheckbox` has to be done in the `LayerGroupState` constructor.

**Review by rldhont — APPROVED** [review]

**Comment by rldhont** [comment]
Thanks @mind84 before merging I have 2 questions 
* Is this PR fixes https://github.com/3liz/lizmap-web-client/issues/3968 and https://github.com/3liz/lizmap-web-client/issues/4232 ?
* Were you thinking of a backport? If yes for which version ?

**Comment by mind84** [comment]
Hi @rldhont,

thank you for taking time to review this PR

> Is this PR fixes https://github.com/3liz/lizmap-web-client/issues/3968 and https://github.com/3liz/lizmap-web-client/issues/4232 ?

Yes, I'll add this information to PR description

> Were you thinking of a backport? If yes for which version ?

Yes, I think this could be backported in 3.8 at least, if you don't see any blocker

Thanks!

File: assets/src/modules/state/Layer.js [hunk 29]
```diff
@@ -44,7 +44,8 @@ export class LayerItemState extends EventDispatcher {
         this._geographicBoundingBox = null;
         this._minScaleDenominator = null;
         this._maxScaleDenominator = null;
-        this._checked = this._parentGroup == null ? true : false;
+        // prop checked is inherithed from Lizmap configuration. if checkboxes are hided for groups then they are checked by default
```
Comment (Gustry) [inline assets/src/modules/state/Layer.js:47]:
```suggestion
        // prop checked is inherited from Lizmap configuration. If checkboxes are hidden for groups then they are checked by default
```

File: assets/src/modules/state/Layer.js [hunk 30]
```diff
@@ -1165,10 +1166,15 @@ export class LayerGroupState extends LayerItemState {
...
      * @param {LayerTreeGroupConfig} layerTreeGroupCfg - the layer item config
      * @param {number[]}             layersOrder       - the layers order
-     * @param {LayerGroupState}     [parentMapGroup]  - the parent layer group
+     * @param {boolean}              hideGroupCheckbox - the option instance from lizMap configuration
+     * @param {LayerGroupState}     [parentMapGroup]   - the hideGroupCheckbox option instance from lizMap configuration
      */
-    constructor(layerTreeGroupCfg, layersOrder, parentMapGroup) {
+    constructor(layerTreeGroupCfg, layersOrder, hideGroupCheckbox, parentMapGroup) {
         super('group', layerTreeGroupCfg, parentMapGroup);
+        // if checkboxes are hided for groups then they are checked by default
```
Comment (Gustry) [inline assets/src/modules/state/Layer.js:1174]:
```suggestion
        // if checkboxes are hidden for groups then they are checked by default
```
https://www.grammarflex.com/whats-the-past-tense-of-hide-is-it-hid-or-hidden/#Forms_of_the_verb_hide

File: tests/end2end/playwright/permalink.spec.js [hunk 31]
```diff
@@ -199,11 +199,16 @@ test.describe('Permalink', () => {
         await expect(hashStyles[3]).toBe(encodeURI('a_single'))
         await expect(hashOpacities[3]).toBe('1')
 
-        // Group as layer has an empty string style ''
-        await expect(hashLayers[4]).toBe(encodeURI('Group as layer'))
-        await expect(hashStyles[4]).toBe('')
+        // legend_option_test is a group, it has no stile
```
Comment (Gustry) [inline tests/end2end/playwright/permalink.spec.js:202]:
```suggestion
        // legend_option_test is a group, it has no style
```

File: tests/end2end/playwright/permalink.spec.js [hunk 32]
```diff
@@ -247,11 +252,16 @@ test.describe('Permalink', () => {
         await expect(hashStyles[3]).toBe(encodeURI('categorized'))
         await expect(hashOpacities[3]).toBe('1')
 
-        // Group as layer has an empty string style ''
-        await expect(hashLayers[4]).toBe(encodeURI('Group as layer'))
-        await expect(hashStyles[4]).toBe('')
+        // legend_option_test is a group, it has no stile
```
Comment (Gustry) [inline tests/end2end/playwright/permalink.spec.js:255]:
```suggestion
        // legend_option_test is a group, it has no style
```

File: tests/end2end/playwright/permalink.spec.js [hunk 33]
```diff
@@ -284,11 +294,16 @@ test.describe('Permalink', () => {
         await expect(hashStyles[1]).toBe(encodeURI('categorized'))
         await expect(hashOpacities[1]).toBe('1')
 
-        // Group as layer has an empty string style ''
-        await expect(hashLayers[2]).toBe(encodeURI('Group as layer'))
-        await expect(hashStyles[2]).toBe('')
+        // legend_option_test is a group, it has no stile
```
Comment (Gustry) [inline tests/end2end/playwright/permalink.spec.js:297]: Same

File: tests/end2end/playwright/permalink.spec.js [hunk 34]
```diff
@@ -322,19 +337,24 @@ test.describe('Permalink', () => {
...
```
Comment (Gustry) [inline tests/end2end/playwright/permalink.spec.js:340]: Same

**Pull request merged**